language-extensions/java/include/JavaArgContainer.h

```diff
@@ -0,0 +1,208 @@
+//*********************************************************************
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+//
+// @File: JavaArgContainer.h
+//
+// Purpose:
+//  Java extension input/output parameter wrappers, along with container
+//  for consolidating them.
+//
+//*********************************************************************
+#pragma once
+
+#include "Common.h"
+
+//---------------------------------------------------------------------
+// Description:
+//	Stores information about the Java argument
+//
+class JavaArg
+{
+	friend class JavaArgContainer;
+
+public:
+	// Constructor to initialize the members
+	//
+	JavaArg(
+		SQLUSMALLINT	id,
+		SQLSMALLINT		type,
+		SQLULEN			size,
+		SQLSMALLINT		decimalDigits,
+		SQLPOINTER		value,
+		SQLINTEGER		strLen_or_Ind,
+		SQLSMALLINT		inputOutputType);
+
+	// Destructor
+	//
+	~JavaArg();
+
+	// Gets id
+	//
+	SQLUSMALLINT GetId() const
+	{
+		return m_id;
+	}
+
+	// Gets type
+	//
+	SQLSMALLINT GetType() const
+	{
+		return m_type;
+	}
+
+	// Gets size
+	//
+	SQLULEN GetSize() const
+	{
+		return m_size;
+	}
+
+	// Gets decimal digits
+	//
+	SQLSMALLINT GetDecimalDigits() const
+	{
+		return m_decimalDigits;
+	}
+
+	// Gets value
+	//
+	SQLPOINTER GetValue() const
+	{
+		return m_value;
+	}
+
+	// Gets string length or null indicator
+	//
+	SQLINTEGER GetStrLenOrInd() const
+	{
+		return m_strLenOrInd;
+	}
+
+	// Gets input/output type
+	//
+	SQLSMALLINT GetInputOutputType() const
+	{
+		return m_inputOutputType;
+	}
+
+private:
+	// Deep copy argument
+	//
+	void DeepCopyValue(
+		SQLSMALLINT			type,
+		SQLINTEGER			size,
+		const SQLPOINTER 	value
+		);
+
+	// Cleans up memory used to store argument
+	//
+	void Cleanup();
+
+	SQLUSMALLINT m_id;				// Id of the argument
+	SQLSMALLINT m_type;				// Data type of the argument
+	SQLULEN m_size;					// Size of the argument
+	SQLSMALLINT m_decimalDigits;	// Decimal digits of the argument
+	SQLPOINTER m_value;				// Value of the argument
+	SQLINTEGER m_strLenOrInd;		// Indicates string length or null
+	SQLSMALLINT m_inputOutputType;	// Argument type (Input/Output)
+};
+
+//---------------------------------------------------------------------
+// Description:
+//	Container for all Java arguments
+//
+class JavaArgContainer
+{
+public:
+	// Constructor
+	//
+	JavaArgContainer()
+	{}
+
+	// Destructor
+	//
+	~JavaArgContainer();
+
+	// Initialize argument container
+	//
+	void Init(const SQLUSMALLINT numOfArgs);
+
+	// Construct and store argument
+	//
+	SQLRETURN AddArg(
```

Comment:
Different tabs? Check the parameter declarations for other functions in this file too.